docs/reference/ilm/getting-started-ilm.asciidoc

```diff
@@ -1,15 +1,190 @@
+a few that are needed for our example. For starters, we will use the
+<<ilm-put-lifecycle,Put Policy>> API to define our first policy. Lifecycle
+policies are defined in JSON and include specific <<ilm-policy-definition,phases and actions>>.
 
-Show explain API to show current state, but ignore the “step” related info,
-only focus on managed/phase/action
+[source,js]
+------------------------
+PUT _ilm/policy/datastream_policy
+{
+  "policy": {
+    "phases": {
+      "hot": {
+        "actions": {
+          "rollover": {
+            "max_size": "50GB",
+            "max_age": "30d"
+          }
+        }
+      },
+      "delete": {
+        "min_age": "90d",
+        "actions": {
+          "delete": {}
+        }
+      }
+    }
+  }
+}
+------------------------
+// CONSOLE
+// TEST
+
+Here we created the policy called `datastream_policy` which rolls over the index
+being written to after it reaches 50 gigabytes, or it is 30 days old. The rollover will
+occur when any of these conditions is true. After 90 days, our index will be deleted.
+
+=== Applying a policy to our index
+
+There are <<set-up-lifecycle-policy,a few ways>> to associate a
+policy to an index. Since we wish specific settings to be applied to
+the new index created from Rollover, we will set the policy via
+index templates.
+
+
+[source,js]
+-----------------------
+PUT _template/datastream_template
+{
+  "index_patterns": ["datastream-*"],
+  "settings": {
+    "number_of_shards": 1,
+    "number_of_replicas": 1,
+    "index.lifecycle.name": "datastream_policy",
+    "index.lifecycle.rollover_alias": "datastream"
+  }
+}
+-----------------------
+// CONSOLE
+// TEST[continued]
+
+The above index template introduces a few new settings specific to {ILM}. The first
+being `index.lifecycle.name`. This setting will configure the "datastream_policy" to the index
+applying this template. Given how index templates work, this means that all newly created
+indices prefixed "datastream-" will be managed by our policy. The other setting used here is
+`index.lifecycle.rollover_alias`. This setting is important to associate which alias to
+rollover on behalf of this index. The intention here is that the rollover alias is also defined
+on the index.
+
+To begin, we will want to bootstrap our first index to write to.
+
+
+[source,js]
+-----------------------
+PUT datastream-00001
+{
+  "aliases": {
+    "datastream": {
+      "is_write_index": true
+    }
+  }
+}
+-----------------------
+// CONSOLE
+// TEST[continued]
+
+When creating our index, we have to consider a few important configurations that
+are tie our index and our policy together correctly. We need to make sure that
+our index name matches our index template pattern of "datastream-*", which it does. We
+are using the rollover action in our policy, which requires that our index name ends
+with a number. In our case, we used `00001`. This is important so that Rollover can
```

Comment:
```suggestion
with a number. In our case, we used `000001`. This is important so that Rollover can
```

As mentioned above, we should use 6-digit examples.